### PR TITLE
Fix junit version

### PR DESCRIPTION
**Pull request by chris-ch**
Along with junit version upgraded to 4.13 (gradle test was failing) and hamcrest updated to 3.0, added Codespace config for easier dev onboarding.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -250,10 +250,10 @@ dependencies {
     implementation "org.unbescape:unbescape:1.1.6.RELEASE"
     implementation "org.slf4j:slf4j-api:1.7.25"
 
-    testImplementation 'junit:junit:4.12'
+    testImplementation 'junit:junit:4.13'
```
Comment (Bombe) [inline build.gradle:253]: Latest JUnit version is 4.13.2. Any particular reason you’re not updating to the latest version?

File: build.gradle [hunk 2]
```diff
@@ -250,10 +250,10 @@ dependencies {
...
     testImplementation "org.mockito:mockito-core:1.9.5"
-    testImplementation "org.hamcrest:hamcrest-library:1.3"
-    testImplementation "org.hamcrest:hamcrest-core:1.3"
+    testImplementation "org.hamcrest:hamcrest-library:3.0"
```
Comment (Bombe) [inline build.gradle:255]: Hamcrest isn’t being split up anymore. Both `org.hamcrest:hamcrest-library` and `org.hamcrest:hamcrest-core` are now basically empty JAR files that both _depend_ on `org.hamcrest:hamcrest` in order to get Maven and Gradle to do the right thing. We should cut out the middleman here.

File: gradle/verification-metadata.xml
Comment (Bombe) [inline gradle/verification-metadata.xml:1]: Hmm, what branch did you base your pull request on? My `gradle/verification-metadata.xml` already contains keys for both hamcrest and JUnit…

File: .vscode/settings.json
Comment (Bombe) [inline .vscode/settings.json:1]: Settings for your local IDE do not belong in version control.

File: .github/dependabot.yml
Comment (Bombe) [inline .github/dependabot.yml:1]: This does not belong in this pull request.

File: .devcontainer/devcontainer.json
Comment (Bombe) [inline .devcontainer/devcontainer.json:1]: This also looks quite unrelated. Please don’t mix unrelated features in pull requests, that just makes it more confusing and more complicated to get anything done.

**Review by Bombe — CHANGES_REQUESTED** [review]

**Comment by chris-ch** [comment]
I re-forked the project, starting from scratch completely and from the "next" branch, the jar build is working:
```
$ ./gradlew jar
```

But then the tests fail:
```
@chris-ch ➜ /workspaces/fred (next) $ ./gradlew --parallel test

> Task :compileTestJava FAILED
/workspaces/fred/test/freenet/client/filter/TheoraBitstreamFilterTest.java:12: error: cannot find symbol
import static org.junit.Assert.assertThrows;
^
  symbol:   static assertThrows
  location: class Assert
/workspaces/fred/test/freenet/support/api/BooleanCallbackTest.java:4: error: cannot find symbol
import static org.junit.Assert.assertThrows;
^
  symbol:   static assertThrows
  location: class Assert
/workspaces/fred/test/freenet/support/SimpleFieldSetTest.java:20: error: cannot find symbol
import static org.hamcrest.Matchers.anEmptyMap;
^
  symbol:   static anEmptyMap
  location: class Matchers
...
```

My understanding is that we should upgrade the junit version to 4.13 from 4.12, and potentially hamcrest as well?

**Pull request closed without merging**